ledger/participant-integration-api/src/main/scala/platform/store/DbSupport.scala

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store
+
+import com.daml.ledger.resources.{ResourceContext, ResourceOwner}
+import com.daml.logging.LoggingContext
+import com.daml.metrics.Metrics
+import com.daml.platform.configuration.ServerRole
+import com.daml.platform.store.appendonlydao.DbDispatcher
+import com.daml.platform.store.backend.StorageBackendFactory
+import com.daml.resources.{PureResource, Resource}
+
+import scala.concurrent.duration.FiniteDuration
+
+case class DbSupport(
+    dbDispatcher: DbDispatcher,
+    storageBackendFactory: StorageBackendFactory,
+)
+
+object DbSupport {
+  def owner(
+      jdbcUrl: String,
+      serverRole: ServerRole,
+      connectionPoolSize: Int,
+      connectionTimeout: FiniteDuration,
+      metrics: Metrics,
+  )(implicit loggingContext: LoggingContext): ResourceOwner[DbSupport] = {
+    val dbType = DbType.jdbcType(jdbcUrl)
+    val storageBackendFactory = StorageBackendFactory.of(dbType)
+    DbDispatcher
+      .owner(
+        dataSource = storageBackendFactory.createDataSourceStorageBackend.createDataSource(jdbcUrl),
+        serverRole = serverRole,
+        connectionPoolSize = connectionPoolSize,
+        connectionTimeout = connectionTimeout,
+        metrics = metrics,
+      )
+      .map(dbDispatcher =>
+        DbSupport(
+          dbDispatcher = dbDispatcher,
+          storageBackendFactory = storageBackendFactory,
+        )
+      )
+  }
+
+  def migratedOwner(
+      jdbcUrl: String,
+      serverRole: ServerRole,
+      connectionPoolSize: Int,
+      connectionTimeout: FiniteDuration,
+      metrics: Metrics,
+  )(implicit loggingContext: LoggingContext): ResourceOwner[DbSupport] = {
+    val migrationOwner = new ResourceOwner[Unit] {
+      override def acquire()(implicit context: ResourceContext): Resource[ResourceContext, Unit] =
+        PureResource(new FlywayMigrations(jdbcUrl).migrate())
+    }
```

Comment:
```suggestion
    val migrationOwner = ResourceOwner.forFuture(() => new FlywayMigrations(jdbcUrl).migrate())
```

Reply:
unfortunately does not work: new FlywayMigrations(...) takes an implicit ResourceContext (to have an EC for doing IO work), which is not available without rolling a ResourceOwner class :cry: 
@SamirTalwar do you know maybe a nice way to be able to propagate the ResourceContext(or the wrapped ec) without implementing ResourceOwner?